.github/workflows/checkmycode.yml

```diff
@@ -1,30 +1,165 @@
-# This workflow will do a clean install of node dependencies, cache/restore them, build the source code and run tests across different versions of node
-# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
-
-name: Node.js CI
-
+name: Testworkflow_CI
 on:
   push:
-    branches: [ main ]
+    branches:
+      - dev
+      - feature    
   pull_request:
-    branches: [ main ]
-
+  release:
+    types: [published, unpublished, created, edited, deleted, prereleased]     
 jobs:
-  build:
-
-    runs-on: ubuntu-latest
-
+  checkout_and_install:
     strategy:
       matrix:
-        node-version: [16.x]
-        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
-
+        node: ["12.22.1", "14.16.1", "15.14.0", "16.1.0"]        
+    runs-on: ubuntu-latest    
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: Install dependencies
+        run: npm ci
+      - name: Installing nyc
+        run: npm install nyc --save-dev
+  build_and_test:
+    needs: 
+      - checkout_and_install      
+    runs-on: ubuntu-latest
+    steps:      
+      - name: npm build, and test
+        run: |
+          npm run build --if-present
+          npm test
```

Comment:
Why test before linting? Why not lint before testing?

Reply:
Accepted, and addressed. Lint should be executed before test.